packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -44,19 +45,55 @@ import {
     TOAST_MESSAGE_ADDED_TO_WISHLIST
 } from '../../constants'
 import {rebuildPathWithParams} from '../../utils/url'
-import {useHistory} from 'react-router-dom'
+import {useHistory, useLocation, useParams} from 'react-router-dom'
 import {useToast} from '../../hooks/use-toast'
+import {useAddToCartModalContext} from '../../hooks/use-add-to-cart-modal'
 
-const ProductDetail = ({category, product, isLoading}) => {
+const ProductDetail = () => {
     const {formatMessage} = useIntl()
-    const basket = useBasket()
     const history = useHistory()
+    const location = useLocation()
     const einstein = useEinstein()
-    const variant = useVariant(product)
     const toast = useToast()
     const navigate = useNavigation()
-    const [primaryCategory, setPrimaryCategory] = useState(category)
+    const {onOpen: onAddToCartModalOpen} = useAddToCartModalContext()
 
+    /****************************** Basket *********************************/
+    const {hasBasket, basket} = useBasket()
+    const createBasket = useShopperBasketsMutation({action: 'createBasket'})
+    const addItemToBasketAction = useShopperBasketsMutation({action: 'addItemToBasket'})
+
+    const {res} = useServerContext()
+    if (res) {
+        res.set('Cache-Control', `max-age=${MAX_CACHE_AGE}`)
+    }
+
+    /*************************** Product Detail and Category ********************/
+    const {productId} = useParams()
+    const urlParams = new URLSearchParams(location.search)
+    const {data: product, isLoading: isProductLoading} = useProduct(
+        {
+            id: urlParams.get('pid') || productId,
+            allImages: true
+        },
+        {
+            keepPreviousData: true
```

Comment:
How about add a comment? It's not immediately clear what the use case is.
```suggestion
            // So that when shoppers select a different variant (and the app fetches the new data),
            // the old data is still rendered (and not the skeletons).
            keepPreviousData: true
```